sockets/utils/formatUrl.js

```diff
@@ -0,0 +1,12 @@
+const url = require('native-url');
+/**
+ * @param {import('./getSocketUrlParts').SocketUrlParts} urlParts
+ * @returns {string}
+ */
+module.exports = function formatUrl(urlParts) {
+  const myUrl = new URL('http://./');
+  for (const [key, value] of Object.entries(urlParts)) {
+    myUrl[key] = value;
```

Comment:
Since the type of `hostname` is `string`, so it's safe to assign its value to `myUrl.hostname`, otherwise we would get result like `http://null`.